src/components/Header/Header.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { postTodo } from '../../api/todos';
+import { formatTodo } from '../../utils/formatResponse';
+
+type Props = {
+  todosToRender: Todo[];
+  setTodosToRender: (arr: Todo[]) => void;
+  pushError: (title: string) => void;
+  setTempTodo: (todo: Todo | null) => void;
+  setIsTempLoading: (val: boolean) => void;
+  setTodos: (arr: Todo[]) => void;
+};
+
+export const Header: React.FC<Props> = ({
+  todosToRender,
+  setTodosToRender,
```

Comment:
Consider renaming 'pushError' to 'showError' or 'displayError' for better clarity on its purpose.